module/VuFind/src/VuFind/Autocomplete/SolrPrefix.php

```diff
@@ -0,0 +1,175 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * @category VuFind
+ * @package  Autocomplete
+ * @author   Vaclav Rosecky <vaclav.rosecky@mzk.cz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:autosuggesters Wiki
+ */
+namespace VuFind\Autocomplete;
+
+/**
+ * Solr autocomplete module with prefix queries using edge N-gram filter
+ *
+ * This class provides suggestions by using the local Solr index.
+ *
+ * @category VuFind
+ * @package  Autocomplete
+ * @author   Vaclav Rosecky <vaclav.rosecky@mzk.cz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:autosuggesters Wiki
+ */
+class SolrPrefix implements AutocompleteInterface
+{
+    /**
+     * Results manager
+     *
+     * @var \VuFind\Search\Results\PluginManager
+     */
+    protected $resultsManager;
+
+    /**
+     * Search object
+     *
+     * @var \VuFind\Search\Solr\Results
+     */
+    protected $searchObject;
+
+    /**
+     * Search class id
+     *
+     * @var string
+     */
+    protected $searchClassId = 'Solr';
+
+    /**
+     * Autocomplete field
+     *
+     * @var string
+     */
+    protected $autocompleteField;
+
+    /**
+     * Facet field
+     *
+     * @var string
+     */
+    protected $facetField;
+
+    /**
+     * Facet limit, can be overridden in subclasses
+     *
+     * @var int
+     */
+    protected $limit = 10;
+
+    /**
+     * Constructor
+     *
+     * @param \VuFind\Search\Results\PluginManager $results Results plugin manager
+     */
+    public function __construct(\VuFind\Search\Results\PluginManager $results)
+    {
+        $this->resultsManager = $results;
+    }
+
+    /**
+     * Get suggestions
+     *
+     * This method returns an array of strings matching the user's query for
+     * display in the autocomplete box.
+     *
+     * @param string $query The user query
+     *
+     * @return array        The suggestions for the provided query
+     */
+    public function getSuggestions($query)
+    {
+        if (!is_object($this->searchObject)) {
+            throw new \Exception('Please set configuration first.');
+        }
+
+        $results = [];
+        try {
+            $params = $this->searchObject->getParams();
+            $rawQuery = $this->autocompleteField . ':(' .
+                $this->mungeQuery($query) . ')';
+            $params->setBasicSearch($rawQuery);
+            $params->addFacet($this->facetField);
+            $params->setLimit(0);
+            $params->setFacetLimit($this->limit);
+            $this->searchObject->getResults();
```

Comment:
One more thing: you could optimize the query a bit by disabling extra functionality:
```
            $options = $params->getOptions();
            $options->disableHighlighting();
            $options->spellcheckEnabled(false);
```